File: tutorials/anchor-browser-agent/data-entry-example.js
let chromium;

try {
    chromium = require('playwright-core').chromium;
} catch (error) {
    if (error.code === 'MODULE_NOT_FOUND') {
        console.log("📦 Missing dependencies detected!");
        console.log("Please install the required packages:");
        console.log("npm install playwright-core");
        process.exit(1);
    } else {
        throw error;
    }
}

const ANCHOR_API_KEY = process.env.ANCHOR_API_KEY;

if (!ANCHOR_API_KEY) {
    throw new Error("ANCHOR_API_KEY is not set, please run `export ANCHOR_API_KEY=<your-api-key>` and try again.");
}

async function executeFormSubmission(connectionString = `wss://connect.anchorbrowser.io?apiKey=${ANCHOR_API_KEY}`) {
    const browser = await chromium.connectOverCDP(connectionString);
    const context = browser.contexts()[0];
    const ai = context.serviceWorkers()[0];
    const page = context.pages()[0];
    
    // Navigate to resume
    await page.goto('https://www.wix.com/demone2/nicol-rider');
    
    // Execute AI task
    const taskPayload = {
        "prompt": "Read the resume, understand the details, and complete the form at https://formspree.io/library/donation/charity-donation-form/preview.html as if you were her. Limit the donation to $10.",
        "provider": "openai",
        "model": "gpt-4",
        "highlight_elements": true
    };
    
    const result = await ai.evaluate(JSON.stringify(taskPayload));
    console.log("Task result:", result);
    await browser.close();
    return result;
}

(async () => {
    try {
        const result = await executeFormSubmission();
        console.log(result);
    } catch (error) {
        console.error("Error:", error.message);
    }
})(); 

File: tutorials/anchor-browser-agent/data-entry-example.py
import os
import json
from playwright.sync_api import sync_playwright

ANCHOR_API_KEY = os.getenv("ANCHOR_API_KEY")

if not ANCHOR_API_KEY:
    raise Exception("ANCHOR_API_KEY is not set, please run `export ANCHOR_API_KEY=<your-api-key>` and try again.")

def execute_form_submission(connection_string= f"wss://connect.anchorbrowser.io?apiKey={ANCHOR_API_KEY}"):
    with sync_playwright() as p:
        browser = p.chromium.connect_over_cdp(connection_string)
        context = browser.contexts[0]
        ai = context.service_workers[0]
        page = context.pages[0]
        # Navigate to resume
        page.goto('https://www.wix.com/demone2/nicol-rider')
        # Execute AI task
        task_payload = {
            "prompt": "Read the resume, understand the details, and complete the form at https://formspree.io/library/donation/charity-donation-form/preview.html as if you were her. Limit the donation to $10.",
            "provider": "openai",
            "model": "gpt-4",
            "highlight_elements": True
        }
        result = ai.evaluate(json.dumps(task_payload))
        print("Task result:", result)
        browser.close()
        return result


print(execute_form_submission())